### ISSUE TITLE
Data Generator (-ing)

### ISSUE THREAD

**Issue by jayjay-cpu** [jayjay-cpu/audio_classification#3]
### Data Generator

------

https://stanford.edu/~shervine/blog/keras-how-to-generate-data-on-the-fly

1.  ` partition  ` dictionary를 만들어준다.

이 dictionary에는 다음과 같은 요소가 들어간다:

- `partition['train']` : a list of training IDs
- `partition['validation']` : a list of validation IDs

2. `labels` dictionary를 만들어준다.

(데이터 셋의) 각각의 `ID`  와 이와 연관된  `labels[ID]`가 들어간다.

<pre><code>
# Training dataset 
ID: { 장발산, 산기슭, 슭곰발} , labels: {1,2,3}
# Validation dataset 
ID: {발냄새} , labels: {1}
# results
partition= {'train': [장발산, 산기슭, 슭곰발], 'validation': [발냄새]}
labels= {장발산:1, 산기슭:2, 슭곰발:3, 발냄새:1} </code></pre>

3. Also, for the sake of **modularity**, we will write Keras code and customized classes in separate files, so that your folder looks like

```
folder/
├── my_classes.py
├── keras_script.py
└── data/
```

where `data/` is assumed to be the folder containing your dataset.

4. Data Generator

   : used for real-time data feeding to your Keras model.

   (1) initialization function of the class 

    We make the latter inherit the properties of `keras.utils.Sequence` so that we can leverage nice functionalities such as *multiprocessing*.

   ```
   def __init__(self, list_IDs, labels, batch_size=32, dim=(32,32,32), n_channels=1,
                n_classes=10, shuffle=True):
       'Initialization'
       self.dim = dim
       self.batch_size = batch_size
       self.labels = labels
       self.list_IDs = list_IDs
       self.n_channels = n_channels
       self.n_classes = n_classes
       self.shuffle = shuffle
       self.on_epoch_end()
   ```
